services/miner/cpuminer.go

```diff
@@ -340,25 +367,85 @@ func (m *CPUMiner) solveBlock(msgBlock *types.Block, ticker *time.Ticker, quit c
 			default:
 				// Non-blocking select to fall through
 			}
-
+			instance := pow.GetInstance(pow.BLAKE2BD,0,[]byte{})
+			powStruct := instance.(*pow.Blake2bd)
 			// Update the nonce and hash the block header.
-			header.Nonce = i
-			h := header.BlockHash()
+			powStruct.Nonce = i
+
+			header.Pow = powStruct
+
 			// Each hash is actually a double hash (tow hashes), so
 			// increment the number of hashes by 2
 			hashesCompleted += 2
+			h := hash.DoubleHashH(header.BlockData())
+			hashNum := pow.HashToBig(&h)
 
+			if hashNum.Cmp(target) > 0 {
```

Comment:
why add the additional debug logic to slow down the calculation instead of the following pseudocode original
```golang
if cmp(hash) <= 0 {
   return true 
} 
```